### PR TITLE
Add trace reporter

### PR DESCRIPTION
**Pull request by stoiev**
Adding trace reporter for DataDog

### CODE REVIEW

File: src/main/scala/kamon/datadog/DatadogSpanReporter.scala [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    val error: Boolean = false
+    new DdSpan(traceId, spanId, parentId, name, resource, service, "custom", start, duration, meta, error)
+
+  }
+}
+
+class DatadogSpanReporter extends SpanReporter {
+  private val translator: KamonDataDogTranslator = KamonDataDogTranslatorDefault$
+  private val logger = LoggerFactory.getLogger(classOf[DatadogAPIReporter])
+  final private val httpConfigPath = "kamon.datadog.trace.http"
```
Comment (Evnsan) [inline src/main/scala/kamon/datadog/DatadogSpanReporter.scala:64]: why we need `final`here?

File: src/test/scala/kamon/datadog/DatadogSpanReporterSpec.scala [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+class DatadogSpanReporterSpec extends AbstractHttpReporter with Matchers with Reconfigure with TestData {
+
+
+  "the DatadogSpanReporter" should {
+    val reporter = new DatadogSpanReporter()
+
+    reporter.start()
+
+    testMap.foreach {
+      case (name, (spans, json)) => {
```
Comment (Evnsan) [inline src/test/scala/kamon/datadog/DatadogSpanReporterSpec.scala]: Do we need this `case` here?

File: src/test/scala/kamon/datadog/DatadogAPIReporterSpec.scala [hunk 3]
```diff
@@ -2,18 +2,27 @@ package kamon.datadog
...
+
+    reporter.start()
 
-    "send counter metrics" in {
-      val reporter = new DatadogAPIReporter()
-      val now = Instant.ofEpochMilli(1523395554)
-      val series = reporter.buildRequestBody(
+    "sends counter metrics" in {
+      val baseUrl = mockResponse("/test", new MockResponse().setStatus("HTTP/1.1 200 OK"))
+      reporter.reconfigure(ConfigFactory.parseString("kamon.datadog.http.api-url = \"" + baseUrl + "\"").withFallback(Kamon.config()))
```
Comment (Evnsan) [inline src/test/scala/kamon/datadog/DatadogAPIReporterSpec.scala]: Could we use applyConfig from kamon here?

File: src/test/scala/kamon/datadog/DatadogSpanReporterSpec.scala [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+        }
+      }
+    }
+
+    s"ignore error responses" in {
+      val baseUrl = mockResponse("/test", new MockResponse().setStatus("HTTP/1.1 500 Internal Server Error"))
+      applyConfig("kamon.datadog.trace.http.api-url = \"" + baseUrl + "\"")
+      reporter.reconfigure(Kamon.config())
+      reporter.reportSpans(testMap.get("single span").get._1)
```
Comment (Evnsan) [inline src/test/scala/kamon/datadog/DatadogSpanReporterSpec.scala]: Could we expand this get to something like this:
```
val (firstSpan, _) = testMap.get("single span").head
reporter.reportSpans(firstSpan)
```

File: src/test/scala/kamon/datadog/DatadogAPIReporterSpec.scala [hunk 5]
```diff
@@ -36,8 +37,8 @@ class DatadogAPIReporterSpec extends AbstractHttpReporter with Matchers {
         )
       )
       val request = server.takeRequest()
-      request.getBody().readUtf8() shouldBe """{"series":[{"metric":"test.counter","interval":1,"points":[[1523394,0]],"type":"count","host":"test","tags":["service:kamon-application","env:staging","tag1:value1"]}]}"""
       request.getRequestUrl.toString shouldEqual baseUrl + "?api_key="
```
Comment (Evnsan) [inline src/test/scala/kamon/datadog/DatadogAPIReporterSpec.scala]: Could we use a dummy key to test this flow?

File: src/test/scala/kamon/datadog/DatadogSpanReporterSpec.scala [hunk 6]
```diff
@@ -90,22 +86,19 @@ trait TestData extends SpanBuilding {
...
 
-
   val otherContextSpan = createSpanContext()
 
   val otherTraceSpan = span.copy(context = otherContextSpan)
   var otherTraceJson = json ++ (Json.obj(
-    "trace_id" -> BigInt(otherContextSpan.traceID.string, 16),
-    "span_id" -> BigInt(otherContextSpan.spanID.string, 16),
-    "parent_id" -> BigInt(otherContextSpan.parentID.string, 16)
+    "trace_id" -> JsNumber(BigDecimal(BigInt(otherContextSpan.traceID.string, 16))),
```
Comment (Evnsan) [inline src/test/scala/kamon/datadog/DatadogSpanReporterSpec.scala]: We should use the same pattern as the one used in line 42

File: src/main/scala/kamon/datadog/DatadogSpanReporter.scala (hunk 1 above)
Comment (stoiev) [reply src/main/scala/kamon/datadog/DatadogSpanReporter.scala:64]: This seems to be the closest to `static private final` from Java that we can get.

`final` is the variable modifier used in [constant expressions](https://www.scala-lang.org/files/archive/spec/2.12/04-basic-declarations-and-definitions.html#value-declarations-and-definitions ), and it generates a [distinct bytecode](https://stackoverflow.com/questions/13412386/why-are-private-val-and-private-final-val-different)

File: src/test/scala/kamon/datadog/DatadogAPIReporterSpec.scala (hunk 3 above)
Comment (stoiev) [reply src/test/scala/kamon/datadog/DatadogAPIReporterSpec.scala]: Already using on next commits

File: src/test/scala/kamon/datadog/DatadogAPIReporterSpec.scala (hunk 5 above)
Comment (stoiev) [reply src/test/scala/kamon/datadog/DatadogAPIReporterSpec.scala]: Done :+1:

**Review by Evnsan — APPROVED** [review]

**Pull request merged**